### PR TITLE
zig build: change "-Drelease" to "-Doptimize"

### PR DESCRIPTION
**Pull request by motiejus**
I find myself quite often creating ReleaseSafe builds and putting them to production for certain experiments:
- Debug info are for stack traces. An ongoing example where those would help is #14815.
- Safety checks would have saved a couple of mine and @kubkon's hours in #15098.

This is a breaking change for scripts that make Zig releases -- I will submit another PR to zig-bootstrap and release-cutter if/after this is merged.

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -22,10 +21,10 @@ pub fn build(b: *std.Build) !void {
...
         break :t b.standardTargetOptions(.{ .default_target = default_target });
     };
-    const optimize: std.builtin.OptimizeMode = if (release) switch (target.getCpuArch()) {
-        .wasm32 => .ReleaseSmall,
-        else => .ReleaseFast,
-    } else .Debug;
+
+    // TODO remove type annotation with ziglang/zig#13749
+    const optimize: std.builtin.Mode = b.option(std.builtin.Mode, "optimize", "Prioritize performance, safety, or binary size (-O flag)") orelse
+        if (target.getCpuArch() == .wasm32) .ReleaseSmall else .Debug;
```
Comment (motiejus) [inline build.zig:27]: question for reviewers: should we port this logic here?

https://github.com/ziglang/zig/blob/8853005672374ceac8760548027a03fa284bf2d1/lib/std/Build.zig#L902-L906

I.e. perhaps everyone who uses `standardOptimizeOption` in wasm32 should get a `ReleaseSmall` if unspecified? cc @Luukdegram

Comment (Luukdegram) [reply build.zig:27]: No, the default should remain `Debug` when unspecified. In `ReleaseSmall` the user would lose all DWARF support when testing their WebAssembly code. Building a release version should be opt-in, not the default.

Comment (motiejus) [reply build.zig:27]: The current default is `ReleaseSmall`. I will change it to `Debug` then. Thanks

Comment (Luukdegram) [reply build.zig:27]: I was specifically answering the question about porting the logic to std's `Build.zig`. Though my answer counts for Zig too, release should remain an opt-in, as it used to.

Comment (motiejus) [reply build.zig]: Great! That simplifies the `build.zig` then. Thanks!

**Pull request merged**